### PR TITLE
[bitnami/postgresql-ha] Add labels and init DB scripts secret

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description of the change**

- Adds labels and podLabels to pgool and postgresql
- Adds initdbScriptsSecret to pgool and postgresql
- Documents pre-existing initdbScripts and initdbScriptsCM for pgpool

**Benefits**

- Brings existing features from postgresql chart to postgresql-ha
- Ability to add custom labels for use in network policies, etc.
- Ability to store sensitive initialization information like passwords in secrets

**Possible drawbacks**

**Applicable issues**

**Additional information**

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

### CODE REVIEW

File: bitnami/postgresql-ha/templates/_helpers.tpl [hunk 1]
```diff
@@ -546,6 +546,13 @@ Return the PostgreSQL initdb scripts configmap.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Get the initialization scripts Secret name.
+*/}}
+{{- define "postgresql-ha.postgresqlInitdbScriptsSecret" -}}
+{{- printf "%s" (tpl .Values.postgresql.initdbScriptsSecret $) -}}
```
Comment (dani8art) [inline bitnami/postgresql-ha/templates/_helpers.tpl:553]: could you reuse `"postgresql-ha.tplValue"`?

File: bitnami/postgresql-ha/templates/_helpers.tpl [hunk 2]
```diff
@@ -557,6 +564,13 @@ Return the Pgpool initdb scripts configmap.
 {{- end -}}
 {{- end -}}
 
+{{/*
+Get the pgpool initialization scripts Secret name.
+*/}}
+{{- define "postgresql-ha.pgpoolInitdbScriptsSecret" -}}
+{{- printf "%s" (tpl .Values.pgpool.initdbScriptsSecret $) -}}
```
Comment (dani8art) [inline bitnami/postgresql-ha/templates/_helpers.tpl:571]: could you reuse `"postgresql-ha.tplValue"`?

File: bitnami/postgresql-ha/templates/pgpool/deployment.yaml [hunk 3]
```diff
@@ -4,6 +4,9 @@ metadata:
   name: {{ include "postgresql-ha.pgpool" . }}
   labels: {{ include "postgresql-ha.labels" . | nindent 4 }}
     app.kubernetes.io/component: pgpool
+    {{- with .Values.pgpool.labels }}
+    {{- toYaml . | nindent 4 }}
```
Comment (dani8art) [inline bitnami/postgresql-ha/templates/pgpool/deployment.yaml:8]: Please, could you use `"postgresql-ha.tplValue"`? and have it as a template value too.

File: bitnami/postgresql-ha/templates/pgpool/deployment.yaml [hunk 4]
```diff
@@ -13,6 +16,9 @@ spec:
     metadata:
       labels: {{ include "postgresql-ha.labels" . | nindent 8 }}
         app.kubernetes.io/component: pgpool
+        {{- with .Values.pgpool.podLabels }}
+        {{- toYaml . | nindent 8 }}
```
Comment (dani8art) [inline bitnami/postgresql-ha/templates/pgpool/deployment.yaml:20]: Please, could you use `"postgresql-ha.tplValue"`? and have it as a template value too.

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 5]
```diff
@@ -4,6 +4,9 @@ metadata:
   name: {{ include "postgresql-ha.postgresql" . }}
   labels: {{- include "postgresql-ha.labels" . | nindent 4 }}
     app.kubernetes.io/component: postgresql
+    {{- with .Values.postgresql.labels }}
+    {{- toYaml . | nindent 4 }}
```
Comment (dani8art) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:8]: Please, could you use `"postgresql-ha.tplValue"`? and have it as a template value too.

File: bitnami/postgresql-ha/templates/postgresql/statefulset.yaml [hunk 6]
```diff
@@ -19,6 +22,9 @@ spec:
     metadata:
       labels: {{- include "postgresql-ha.labels" . | nindent 8 }}
         app.kubernetes.io/component: postgresql
+        {{- with .Values.postgresql.podLabels }}
+        {{- toYaml . | nindent 8 }}
```
Comment (dani8art) [inline bitnami/postgresql-ha/templates/postgresql/statefulset.yaml:26]: Please, could you use `"postgresql-ha.tplValue"`? and have it as a template value too.

File: bitnami/postgresql-ha/values-production.yaml [hunk 7]
```diff
@@ -121,6 +121,14 @@ clusterDomain: cluster.local
 ## PostgreSQL parameters
 ##
 postgresql:
+  ## Labels to add to the StatefulSet
```
Comment (dani8art) [inline bitnami/postgresql-ha/values-production.yaml:124]: Add `evaluated as template`, please.

File: bitnami/postgresql-ha/values-production.yaml [hunk 8]
```diff
@@ -121,6 +121,14 @@ clusterDomain: cluster.local
...
+  ##
+  labels: {}
+
+  ## Labels to add to the StatefulSet pods
```
Comment (dani8art) [inline bitnami/postgresql-ha/values-production.yaml:128]: Add `evaluated as template`, please.

File: bitnami/postgresql-ha/values-production.yaml [hunk 9]
```diff
@@ -121,6 +121,14 @@ clusterDomain: cluster.local
...
+  ##
+  podLabels: {}
```
Comment (dani8art) [inline bitnami/postgresql-ha/values-production.yaml:130]: Add `evaluated as template`, please.

File: bitnami/postgresql-ha/values-production.yaml [hunk 10]
```diff
@@ -287,9 +295,22 @@ postgresql:
...
+  ## Note: can be used with initdbScriptsCM or initdbScripts
+  ##
+  # initdbScriptsSecret:
+
 ## Pgpool parameters
 ##
 pgpool:
+  ## Labels to add to the Deployment
+  ##
+  labels: {}
```
Comment (dani8art) [inline bitnami/postgresql-ha/values-production.yaml:308]: Add `evaluated as template`, please.

File: bitnami/postgresql-ha/values-production.yaml [hunk 11]
```diff
@@ -287,9 +295,22 @@ postgresql:
...
+
+  ## Labels to add to the pods
+  ##
+  podLabels: {}
```
Comment (dani8art) [inline bitnami/postgresql-ha/values-production.yaml:312]: Add `evaluated as template`, please.

File: bitnami/postgresql-ha/values.yaml [hunk 12]
```diff
@@ -121,6 +121,14 @@ clusterDomain: cluster.local
 ## PostgreSQL parameters
 ##
 postgresql:
+  ## Labels to add to the StatefulSet
+  ##
+  labels: {}
```
Comment (dani8art) [inline bitnami/postgresql-ha/values.yaml:126]: Add `evaluated as template`, please.

File: bitnami/postgresql-ha/values.yaml [hunk 13]
```diff
@@ -121,6 +121,14 @@ clusterDomain: cluster.local
...
+
+  ## Labels to add to the StatefulSet pods
+  ##
+  podLabels: {}
```
Comment (dani8art) [inline bitnami/postgresql-ha/values.yaml:130]: Add `evaluated as template`, please.

File: bitnami/postgresql-ha/values.yaml [hunk 14]
```diff
@@ -287,9 +295,22 @@ postgresql:
...
+  ## Note: can be used with initdbScriptsCM or initdbScripts
+  ##
+  # initdbScriptsSecret:
+
 ## Pgpool parameters
 ##
 pgpool:
+  ## Labels to add to the Deployment
+  ##
+  labels: {}
```
Comment (dani8art) [inline bitnami/postgresql-ha/values.yaml:308]: Add `evaluated as template`, please.

File: bitnami/postgresql-ha/values.yaml [hunk 15]
```diff
@@ -287,9 +295,22 @@ postgresql:
...
+
+  ## Labels to add to the pods
+  ##
+  podLabels: {}
```
Comment (dani8art) [inline bitnami/postgresql-ha/values.yaml:312]: Add `evaluated as template`, please.

**Comment by steven-sheehy** [comment]
@dani8art Thanks for the review. I think I've addressed all your feedback if you can take another look.

**Comment by steven-sheehy** [comment]
Done

**Comment by juan131** [comment]
Waiting for CI/CD

**Pull request merged**